src/wp-includes/cron.php

```diff
@@ -235,7 +317,17 @@ function wp_schedule_event( $timestamp, $recurrence, $hook, $args = array() ) {
 		'interval' => $event->interval,
 	);
 	uksort( $crons, 'strnatcasecmp' );
-	return _set_cron_array( $crons );
+
+	$set = _set_cron_array( $crons );
+
+	if ( $wp_error && ! $set ) {
```

Comment:
Do you think `wp_schedule_event()` _should_ have a duplicate check in that case? Then it can early return either a `WP_Error` or boolean false depending on `$wp_error`.

Reply:
There is a [ticket for a duplicate check of recurring events](https://core.trac.wordpress.org/ticket/49693) on trac. I'm not convinced it needs it but it seems I'm in the minority. I've left some feedback on that ticket for the current patch if it's to go in.